Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs

```diff
@@ -143,13 +150,21 @@ updateLocationHandler ::
     HasFlowEnv m r '["driverLocationUpdateTopic" ::: Text],
     MonadTime m,
     CacheFlow m r,
-    EsqDBFlow m r
+    EsqDBFlow m r,
+    Esq.EsqDBReplicaFlow m r
   ) =>
   UpdateLocationHandle m ->
   UpdateLocationReq ->
   m APISuccess
 updateLocationHandler UpdateLocationHandle {..} waypoints = withLogTag "driverLocationUpdate" $
   withLogTag ("driverId-" <> driver.id.getId) $ do
+    lastPaymentOverdue <- runInReplica $ findOldestFeeByStatus (cast driver.id) PAYMENT_OVERDUE
```

Comment:
As update location API is being called again and again, so its for now just have this check only for merchant which has `subscription` true in merchant config.